orangecontrib/timeseries/widgets/owdifference.py

```diff
@@ -70,17 +88,30 @@ def __init__(self):
         view.selectionModel().selectionChanged.connect(self.on_changed)
         box.layout().addWidget(view)
         gui.auto_commit(box, self, 'autocommit', '&Apply')
+        self.combo_changed()
 
     @Inputs.time_series
     def set_data(self, data):
-        self.data = data = None if data is None else Timeseries.from_data_table(data)
         if data is not None:
+            data = Timeseries.from_data_table(data)
             self.model.wrap([var for var in data.domain.variables
                              if var.is_continuous and var is not data.time_variable])
+        self.data = data
+        self.on_changed()
+
+    def combo_changed(self):
+        if self.chosen_operation == 0:
+            self.operation = self.Operation.DIFF
+        elif self.chosen_operation == 1:
+            self.operation = self.Operation.QUOT
+        else:
+            self.operation = self.Operation.PERC
```

Comment:
Whenever you see something like this, cringe. And then preferably start over. :laughing: